File: Dave Min/2023-02-03-ReorderList.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
    public void ReorderList(ListNode head)
    {
        ListNode root = head;
        List<int> list = new List<int>();
        while (root != null)
        {
            list.Add(root.val);
            root = root.next;
        }
        root = head;
        for (int i = 0; i < list.Count; i++)
        {
            if (i % 2 == 0)
                root.val = list[i / 2];
            else
                root.val = list[list.Count - 1 - i / 2];
            root = root.next;
        }

    }
}

File: MinjiJu/2023-02-03-Reorder-List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    void reorderList(ListNode* head) {
        
        if(!head || !(head->next) || !(head->next->next)) return;
        
        ListNode* last = head;
        while(last->next->next) last = last->next;
        
        last->next->next = head->next;
        head->next = last->next;
        last->next = NULL;
        reorderList(head->next->next);
    }
};